web/modules/custom/liiweb_legislation/liiweb_legislation.module

```diff
@@ -179,67 +179,41 @@ function liiweb_legislation_theme() {
  */
 function liiweb_legislation_node_view(array &$build, \Drupal\Core\Entity\EntityInterface $entity, \Drupal\Core\Entity\Display\EntityViewDisplayInterface $display, $view_mode) {
   if($entity->getType() == 'legislation' && $view_mode == 'full') {
-    $raw_json = json_decode($build['#node']->get('field_raw_json')->getValue()[0]['value']);
-    $build['#attached']['drupalSettings']['liiweb_legislation']['raw_json'] = $raw_json;
+    /** @var \Drupal\liiweb\LiiWebUtils $utils */
+    $utils = \Drupal::service('liiweb.utils');
+    /** @var \Drupal\liiweb_legislation\LiiWebLegislationUIHelper $legislationUtils */
+    $legislationUtils = \Drupal::service('liiweb_legislation.ui.utils');
+    /** @var \Drupal\node\Entity\Node $node */
+    $node = $build['#node'];
+    $expressionJson = $utils->getLegislationJsonData($node);
+    $build['#attached']['drupalSettings']['liiweb_legislation']['raw_json'] = $expressionJson;
     // stash the country code for use when displaying the legislation content
-    $build['field_content']['country'] = $raw_json->country;
+    $build['field_content']['country'] = $expressionJson->country;
 
     try {
       /** @var \Drupal\node\Entity\Node $repeal */
-      $repeal = $build['#node']->field_repeal->entity->field_repeal_work->entity;
+      $repeal = $node->field_repeal->entity->field_repeal_work->entity;
     } catch (\Exception $e) {
       $repeal = NULL;
     }
-    $unknown = new TranslatableMarkup('Unknown');
     if($repeal) {
-      $repeal_date = $unknown;
-      if ($date = $build['#node']->field_repeal->entity->field_repeal_date->date) {
-        $repeal_date = \Drupal::service('date.formatter')->format(
-          $date->getTimestamp(), 'custom', 'j F, Y'
-        );
-      }
-      try {
-        $link = Link::createFromRoute($repeal->getTitle(),
-          'entity.node.canonical', ['node' => $repeal->id()]
-        )->toString();
-      } catch (\Exception $e) {
-        $link = $unknown;
-      }
+      $repealDate = $node->field_repeal->entity->field_repeal_date->date;
       $build['field_frbr_uri']['#attributes']['class'][] = 'alert-red';
       $build['field_frbr_uri'][0]['#template'] = '{{ value|raw }}';
-      $build['field_frbr_uri'][0]['#context']['value'] = new TranslatableMarkup(
-        'This legislation was repealed on <strong>@date</strong> by <h3 class="h6">@link</h3>',
-        [
-          '@date' => $repeal_date,
-          '@link' => $link
-        ]
-      );
+      $build['field_frbr_uri'][0]['#context']['value'] = $legislationUtils->formatBannerRepealedExpression($repeal, $repealDate);
     }
-    elseif($build['#node']->isDefaultRevision()) {
+    elseif($node->isDefaultRevision()) {
       $build['field_frbr_uri']['#attributes']['class'][] = 'alert-primary';
-      $build['field_frbr_uri'][0]['#context']['value'] = new TranslatableMarkup('This is the latest version of this legislation.');
+      $from = $expressionJson->expression_date;
```

Comment:
This should be:

`$from = $expressionJson->commencement_date`

but it may be null.